config_test.go

```diff
@@ -41,6 +44,69 @@ func TestDefaultConfig(t *testing.T) {
 	checkDefault(t, &config)
 }
 
+func TestConfig_SetLabels(t *testing.T) {
+	config := hazelcast.NewConfig()
+	m := sync.Mutex{}
+	for _, tc := range []struct {
+		info           string
+		expectedLength int
+		input          []string
+	}{
+		{"non-empty single string slice", 1, []string{"client-label"}},
+		{"empty single string slice", 1, []string{""}},
+		{"empty slice", 0, []string{}},
+		{"non-empty multiple string slice", 2, []string{"a", "b"}},
+		{"hybrid strings slice", 3, []string{"a", "", "c"}},
+	} {
+		t.Run(tc.info, func(t *testing.T) {
+			m.Lock()
+			defer func(config *hazelcast.Config) {
+				*config = hazelcast.NewConfig()
+			}(&config)
+			defer m.Unlock()
+			config.SetLabels(tc.input...)
+			got := len(config.Labels)
+			if got != tc.expectedLength {
+				t.Fatalf("got %v want %v", got, tc.expectedLength)
+			}
+			labels := config.Labels
+			for i := 0; i < got; i++ {
+				if labels[i] != tc.input[i] {
+					t.Fatalf("got %v want %v", labels[i], tc.input[i])
+				}
+			}
+		})
+	}
+}
+
+func TestConfig_Clone(t *testing.T) {
+	cfg := hazelcast.Config{
+		FlakeIDGenerators: map[string]hazelcast.FlakeIDGeneratorConfig{
+			"test-flakeID-key-1": {
+				PrefetchCount:  50_000,
+				PrefetchExpiry: types.Duration(time.Minute * 2),
+			},
+			"test-flakeID-key-2": {
+				PrefetchCount:  90_000,
+				PrefetchExpiry: types.Duration(time.Minute * 5),
+			},
+		},
+		Labels:     []string{"test-client-label"},
+		ClientName: "test-client",
+		// Each config's clone method is assumed to working fine by itself.
+		// This test is only validating hazelcast config's clone receiver.
+		Logger:        logger.Config{},
+		Failover:      cluster.FailoverConfig{},
+		Serialization: serialization.Config{},
+		Cluster:       cluster.Config{},
+		Stats:         hazelcast.StatsConfig{},
+	}
+	newCfg := cfg.Clone()
+	assert.True(t, reflect.DeepEqual(newCfg.FlakeIDGenerators, cfg.FlakeIDGenerators))
```

Comment:
Because of the sub-content of the other config which has slice for example, `DeepEqual` fails. If we want to directly use the `DeepEqual` to compare two different config, we need to modify some of the clone method of configs. You can explore the problem right here:
https://github.com/hazelcast/hazelcast-go-client/issues/818

Reply:
Indeed, `DeepEqual` cannot work for entire hazelcast config view and its sub-content in one shot. But this does not mean that clone does not work properly.